### PR TITLE
CPM-476: Add TableInputMeasurement

### PR DESCRIPTION
**Pull request by pierallard**
Adds TableInputMeasurement

A little refacto for the highlight cell to DRY

Screenshot from 2022-01-19 11-08-42

### CODE REVIEW

File: front-packages/akeneo-design-system/src/components/Input/TableInput/TableInput.stories.mdx [hunk 1]
```diff
@@ -153,16 +133,18 @@ Table input allows the user to input content in a table.
...
-                <TableInput.Cell><TableInput.Select onSearchChange={() => {}} value={'Select value'} highlighted={r === 2}/></TableInput.Cell>
-                <TableInput.Cell><TableInput.Text value={'Text value'} highlighted={r === 3}/></TableInput.Cell>
-                <TableInput.Cell><TableInput.Number value={42} highlighted={r === 4}/></TableInput.Cell>
-                <TableInput.Cell><TableInput.Boolean value={true} yesLabel={"Yes"} noLabel={"No"} highlighted={r === 5}/></TableInput.Cell>
+                <TableInput.Cell><TableInput.CellContent rowTitle={true} highlighted={r === 0} inError={r === 1}>Value</TableInput.CellContent></TableInput.Cell>
+                <TableInput.Cell><TableInput.CellContent highlighted={r === 1} inError={r === 2}>Value</TableInput.CellContent></TableInput.Cell>
+                <TableInput.Cell><TableInput.Select onSearchChange={() => {}} value={'Select value'} highlighted={r === 2} inError={r === 3}/></TableInput.Cell>
+                <TableInput.Cell><TableInput.Text value={'Text value'} highlighted={r === 3}  inError={r === 4}/></TableInput.Cell>
```
Comment (pierallard) [inline front-packages/akeneo-design-system/src/components/Input/TableInput/TableInput.stories.mdx:144]: DOUBLE SPACE

File: front-packages/akeneo-design-system/src/storybook/hooks/useFakeTableInput.ts [hunk 2]
```diff
@@ -8,8 +8,28 @@ type Row = {
   origin: 'french' | 'english' | 'german' | null;
 };
 
-type RowCode = 'nutritionScore' | 'part' | 'quantity' | 'is_allergenic' | 'origin';
-type RowValue = '1' | '2' | '3' | string | number | boolean | null | 'french' | 'english' | 'german';
+type RowCode = 'nutritionScore' | 'part' | 'quantity' | 'is_allergenic' | 'origin' | 'weight';
```
Comment (mmetayer) [inline front-packages/akeneo-design-system/src/storybook/hooks/useFakeTableInput.ts:11]:
```suggestion
type RowCode = 'nutritionScore' | 'part' | 'quantity' | 'is_allergenic' | 'origin' | 'length';
```

File: front-packages/akeneo-design-system/src/storybook/hooks/useFakeTableInput.ts [hunk 3]
```diff
@@ -20,14 +40,15 @@ const useFakeTableInput = (linesCount: number) => {
         quantity: lineIndex * 10,
         is_allergenic: [true, false, null][lineIndex % 3],
         origin: ['french', 'english', 'german', null][lineIndex % 4] as 'french' | 'english' | 'german' | null,
+        weight: {unit: weightUnits[lineIndex].value, amount: `${lineIndex * 1.5}`},
```
Comment (mmetayer) [inline front-packages/akeneo-design-system/src/storybook/hooks/useFakeTableInput.ts:43]:
```suggestion
        length: {unit: lengthUnits[lineIndex].value, amount: `${lineIndex * 1.5}`},
```

File: front-packages/akeneo-design-system/src/storybook/hooks/useFakeTableInput.ts [hunk 4]
```diff
@@ -8,8 +8,28 @@ type Row = {
...
+  | string
+  | number
+  | boolean
+  | null
+  | 'french'
+  | 'english'
+  | 'german'
+  | {unit: string; amount: string};
+
+const weightUnits = [
```
Comment (mmetayer) [inline front-packages/akeneo-design-system/src/storybook/hooks/useFakeTableInput.ts:25]:
```suggestion
const lengthUnits = [
```

File: front-packages/akeneo-design-system/src/storybook/hooks/useFakeTableInput.ts [hunk 5]
```diff
@@ -39,7 +60,7 @@ const useFakeTableInput = (linesCount: number) => {
     return text.toLowerCase().includes(searchValue.toLowerCase());
   };
 
-  return {getValue, setValue, searchValue, setSearchValue, match};
+  return {getValue, setValue, searchValue, setSearchValue, match, weightUnits};
```
Comment (mmetayer) [inline front-packages/akeneo-design-system/src/storybook/hooks/useFakeTableInput.ts:63]:
```suggestion
  return {getValue, setValue, searchValue, setSearchValue, match, lengthUnits};
```

File: front-packages/akeneo-design-system/src/components/Input/TableInput/TableInput.stories.mdx [hunk 6]
```diff
@@ -25,14 +27,15 @@ Table input allows the user to input content in a table.
 <Canvas>
   <Story name="Standard">
     {args => {
-      const {getValue, setValue, searchValue, setSearchValue, match} = useFakeTableInput(6);
+      const {getValue, setValue, searchValue, setSearchValue, match, weightUnits} = useFakeTableInput(6);
```
Comment (mmetayer) [inline front-packages/akeneo-design-system/src/components/Input/TableInput/TableInput.stories.mdx:30]:
```suggestion
      const {getValue, setValue, searchValue, setSearchValue, match, lengthUnits} = useFakeTableInput(6);
```

File: front-packages/akeneo-design-system/src/components/Input/TableInput/TableInput.stories.mdx [hunk 7]
```diff
@@ -25,14 +27,15 @@ Table input allows the user to input content in a table.
...
       return <TableInput {...args}>
         <TableInput.Header>
           <TableInput.HeaderCell>Ingredient</TableInput.HeaderCell>
           <TableInput.HeaderCell>Nutrition Score</TableInput.HeaderCell>
           <TableInput.HeaderCell>For 1 part</TableInput.HeaderCell>
           <TableInput.HeaderCell>Quantity</TableInput.HeaderCell>
           <TableInput.HeaderCell>Is allergenic</TableInput.HeaderCell>
+          <TableInput.HeaderCell>Weight</TableInput.HeaderCell>
```
Comment (mmetayer) [inline front-packages/akeneo-design-system/src/components/Input/TableInput/TableInput.stories.mdx:38]:
```suggestion
          <TableInput.HeaderCell>Length</TableInput.HeaderCell>
```

**Review by amigaire — APPROVED** [review]

File: front-packages/akeneo-design-system/src/components/Input/TableInput/TableInput.stories.mdx [hunk 8]
```diff
@@ -57,6 +60,14 @@ Table input allows the user to input content in a table.
             <TableInput.Cell><TableInput.Text value={getValue(index, 'part')} onChange={(value) => setValue(index, 'part', value)}/></TableInput.Cell>
             <TableInput.Cell><TableInput.Number value={getValue(index, 'quantity')} onChange={(value) => setValue(index, 'quantity', value)}/></TableInput.Cell>
             <TableInput.Cell><TableInput.Boolean yesLabel="Yes" noLabel="No" value={getValue(index, 'is_allergenic')} onChange={(value) => setValue(index, 'is_allergenic', value)}/></TableInput.Cell>
+            <TableInput.Cell>
+              <TableInput.Measurement
+                amount={getValue(index, 'weight').amount}
+                unit={getValue(index, 'weight').unit}
+                units={weightUnits}
+                onChange={(amount, unit) => setValue(index, 'weight', {unit, amount})}
```
Comment (mmetayer) [inline front-packages/akeneo-design-system/src/components/Input/TableInput/TableInput.stories.mdx:68]:
```suggestion
                amount={getValue(index, 'length').amount}
                unit={getValue(index, 'length').unit}
                units={lengthUnits}
                onChange={(amount, unit) => setValue(index, 'length', {unit, amount})}
```

**Review by mmetayer — APPROVED** [review]

**Pull request merged**